### DISCUSSION TITLE
Hope the rules cli could be run step by step (i.e. continue from a given state)

### DISCUSSION THREAD

**Discussion by henryclw** [BattlesnakeOfficial/feedback#197] [Feature Requests]
The current [rules cli](https://github.com/BattlesnakeOfficial/rules/blob/main/cli/README.md) is running from the very beginning of the game. Hope we could input some game state and continue the gaming from that state.

**Comment by robbles** [comment]
Hi @henryclw, thanks for this suggestion!

Are you thinking the game state input would be something like what's passed to the snakes on a move request?

**Reply by henryclw** [reply]
Thank you for you kind reply

I want to pass the state to the [rules cli](https://github.com/BattlesnakeOfficial/rules/blob/main/cli/README.md), so that the game could start from the previous state (e.g. the whole body array of snake 1, the whole body array of snake 2, and so on, also the foods positions). Then the cli would pass these information to the snakes, and get the decision from snakes url, then continue (or resume) the game.

Kind of like the endgame in chess
> In [chess](https://en.wikipedia.org/wiki/Chess) and chess-like games, the endgame (or end game or ending) is the stage of the game when few [pieces](https://en.wikipedia.org/wiki/Chess_piece) are left on the board.

For the time being, we could only start the gaming from a blank state (every time is the same, and needs a lot of time for snakes to grow)
